packages/release-please/src/release-please.ts

```diff
@@ -383,13 +385,42 @@ async function runBranchConfiguration(
       plugins
     );
     try {
-      const numReleases = await Runner.createReleases(manifest);
-      logger.info(`Created ${numReleases} releases`);
-      if (numReleases > 0) {
+      const releases = await Runner.createReleases(manifest);
+      logger.info(`Created ${releases.length} releases`);
+      if (releases.length > 0) {
         // we created a release, reload config which may include the latest
         // version
         manifest = null;
       }
+      if (branchConfiguration.tagPullRequestNumber) {
+        // Record the pull request number to the commit as
+        // a tag.
+
+        // It's possible for manifest.createReleases() to create
+        // releases for multiple pull requests. Find the unique
+        // pair of pull request numbers and their commit SHAs.
+        const prNumberToSha = new Map<number, string>();
+        for (const release of releases) {
+          prNumberToSha.set(release.prNumber, release.sha);
+        }
+        for (const prNumber of prNumberToSha.keys()) {
+          // A Git tag is a ref. A lightweight tag only requires this
+          // create references API call, rather than a tag objec
+          // (/repos/{owner}/{repo}/git/tags).
+          // https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28#create-a-tag-object
+          // https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#create-a-reference
+          const sha = prNumberToSha.get(prNumber)!;
+          const tagRefName = `refs/tags/release-please-${prNumber}`;
+          logger.info(`Creating ${tagRefName} pointing to ${sha}`);
+          const tagResponse = await octokit.git.createRef({
```

Comment:
minor nit that could make the tests cleaner would be to move this into a function like `createTag` where we wrap the octokit call.

Rather than testing the HTTP mock in the event handler test, you can assert that `createTag` is called N times.

Then separately, you could test the implementation of the `createTag` where we mock the HTTP call.

Reply:
(Thinking about which class should have `createTag` method) Is that the `Runner` for? Now it makes sense.

Reply:
Done.